gtk4/src/custom_layout.rs

```diff
@@ -0,0 +1,73 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{CustomLayout, Orientation, SizeRequestMode, Widget};
+use glib::translate::*;
+
+impl CustomLayout {
+    #[doc(alias = "gtk_custom_layout_new")]
+    pub fn new<F, G, H>(request_mode: F, measure: G, allocate: H) -> CustomLayout
+    where
+        F: Fn(&Widget) -> SizeRequestMode,
+        G: Fn(&Widget, Orientation, i32, i32, i32, i32, i32),
+        H: Fn(&Widget, i32, i32, i32),
+    {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gtk_custom_layout_new(
+                Some(request_mode_trampoline::<F>),
+                Some(measure_trampoline::<G>),
+                Some(allocate_trampoline::<H>),
+            ))
+        }
+    }
+}
+
+unsafe extern "C" fn request_mode_trampoline<F>(
+    widget: *mut ffi::GtkWidget,
+    f: glib::ffi::gpointer,
+) -> ffi::GtkSizeRequestMode
+where
+    F: Fn(&Widget) -> SizeRequestMode,
+{
+    let f: &F = &*(f as *const F);
+    f(&from_glib_borrow(widget)).to_glib()
+}
+
+unsafe extern "C" fn measure_trampoline<G>(
+    widget: *mut ffi::GtkWidget,
+    orientation: ffi::GtkOrientation,
+    for_size: *mut i32,
+    minimum: *mut i32,
+    natural: *mut i32,
+    minimum_baseline: *mut i32,
+    natural_baseline: *mut i32,
```

Comment:
These are all `out` parameters, or maybe even `inout`. You need to write the result to them